app/serializers/api/v1/message_serializer.rb

```diff
@@ -3,12 +3,20 @@ class MessageSerializer < ApplicationSerializer
     embed :ids, include: true
 
     attributes :id, :body, :state, :is_private, :created_at,
-      :updated_at, :offer_id, :item_id, :designation_id, :order_id
+      :updated_at, :messageable_type, :messageable_id
 
     has_one :sender, serializer: UserSerializer, root: :user
 
     def designation_id
-      object.order_id
+      object.messageable_type == 'Order' ? object.messageable_id : nil
+    end
+
+    def offer_id
+      object.messageable_type == 'Offer' ? object.messageable_id : nil
+    end
+
+    def item_id
+      object.messageable_type == 'Item' ? object.messageable_id : nil
```

Comment:
Good to keep those for backwards compatibility. Although you probably want to mark them as deprecated in a comment. We probably shouldn't rely on them anywhere in the apps. Better to rely on `messageable`

Reply:
sure. But I think i'll remove them. We can add it if required